standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -8142,9 +8146,11 @@ private void dropPartitionAllColumnGrantsNoTxn(
       query.declareParameters("java.lang.String t1");
       mSecurityDCList = (List<MDCPrivilege>) query.execute(dcName);
     }
+    try (Query q = query) {
```

Comment:
We might have an error during execution. Should we close the query there too?

Reply:
Done